### ISSUE TITLE
download: consider changing asset ordering

### ISSUE THREAD

**Issue by ee7** [ziglang/www.ziglang.org#308]
At the time of writing, the ordering of the Zig 0.11.0 assets (and the Zig master assets, minus FreeBSD) on https://ziglang.org/download/ is:

| OS      | Arch        |
| :------ | :---------- |
| Windows | x86_64      |
| Windows | x86         |
| Windows | aarch64     |
| macOS   | aarch64     |
| macOS   | x86_64      |
| Linux   | x86_64      |
| Linux   | x86         |
| Linux   | aarch64     |
| Linux   | armv7a      |
| Linux   | riscv64     |
| Linux   | powerpc64le |
| Linux   | powerpc     |
| FreeBSD | x86_64      |

Questions:

1. Is the current Arch ordering (for a given OS) deliberate? I am aware of the [support table](https://ziglang.org/download/0.11.0/release-notes.html#Support-Table), where the ordering starts x86_64, x86, aarch64. But, for example, is there a reason that aarch64 is at the top for macOS, but below x86 for Windows and Linux? 
2. Could we order assets for a given OS differently, perhaps alphabetically by architecture?
3. Is the current ordering of OSes deliberate?
4. Could we order the OSes (non-reverse) alphabetically? Or perhaps near-alphabetically, the same as the support table: Linux, macOS, Windows, BSDs.

Even if the above asset ordering is supposed to convey something, the ordering isn't currently explained on that page. And I think on a download page people mostly just want to:

- get a quick sense of the available assets
- click the asset they want

I think people are used to assets being ordered alphabetically (for example, on GitHub - see e.g. [hyperfine assets](https://github.com/sharkdp/hyperfine/releases/tag/v1.17.0)). And I find the below ordering to be significantly clearer (I'm also fine with BSDs being at the bottom). 

Thoughts?

| OS      | Arch        |
| :------ | :---------- |
| FreeBSD | x86_64      |
| Linux   | aarch64     |
| Linux   | armv7a      |
| Linux   | powerpc     |
| Linux   | powerpc64le |
| Linux   | riscv64     |
| Linux   | x86         |
| Linux   | x86_64      |
| macOS   | aarch64     |
| macOS   | x86_64      |
| Windows | aarch64     |
| Windows | x86         |
| Windows | x86_64      |

**Comment by kristoff-it** [comment]
Yes the ordering of OSs and architectures is deliberate and defined here https://github.com/ziglang/www.ziglang.org/blob/master/config.toml#L29-L50

This was originally designed to be like this by Andrew and since I too shared the same preference, I reimplemented the original system when I redesigned the website.

> Even if the above asset ordering is supposed to convey something, the ordering isn't currently explained on that page. 

True, but at the same time the idea is simple:

1. windows is at the top so that, as you scroll towards your OS, you see that we do have support for windows. it's common for OSS projects to leave windows support as an afterthought and this way we silently say that we hope to do better / make sure that more people become carriers of that information

2. arches are sorted by popularity, so that most people will find their arch right away. So that means x86_64 for windows, and aarch64 for macOS

Hopefully this makes clear the intent behind the ordering, and why we never really made it explicit.

In the end, in my opinion, while minor, it's a good thing to have even at the expense of making the sorting a bit weird for people that stop to smell the ordering :^)

I've merged #311 and with that implemented I don't feel impelled to make any other changes given the information at my disposal. That said, happy to discuss it further if you want to add some more arguments to the mix.
